Sage/carbon
# fix(menu-fullscreen): ensure menu has aria properties required for accessible dialogs

closes #5631

### Proposed behaviour

For `MenuFullscreen`:

- Add translation key for the opened menu's close button.
- Add `aria-label` prop to control `aria-label` attribute on root container.
- Ensure opened menu has all [the necessary ARIA attributes set to be classified as a modal](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#wai-ariaroles,states,andproperties), these are `aria-label`, `aria-modal` and `role`.
- Set `aria-modal` attribute on the menu's root container when it is open and is the topmost modal on the page.
- Refactor test code

### Current behaviour

- `MenuFullscreen`, when opened, does not have an ARIA role of dialog, nor the `aria-modal` attribute set.
- No translation support for any of the `aria-label`s in the component's markup

### Checklist




#### QA


### Testing instructions

Using Safari with macOS VoiceOver, Chrome with NVDA and Firefox with NVDA. Please do the following:

1. Do npm start. Once storybook is loaded, open the `fullscreen view` story in canvas mode in a new tab
2. Start screen reader
3. Ensure the width of the viewport is less than 1200px
4. Open menu with the keyboard
5. Verify screen reader announces the `aria-label` of the menu and it is a dialog with content.

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

[Link to codesandbox showing broken behaviour](https://codesandbox.io/s/menu-fullscreen-aria-properties-t3hpc4?file=/src/App.js)
